processing/src/main/java/io/druid/query/filter/JavaScriptDimFilter.java

```diff
@@ -21,32 +21,36 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Joiner;
 import com.google.common.base.Preconditions;
 import com.metamx.common.StringUtils;
 
 import java.nio.ByteBuffer;
+import java.util.Arrays;
 
 public class JavaScriptDimFilter implements DimFilter
 {
-  private final String dimension;
+  private final String[] dimensions;
   private final String function;
 
   @JsonCreator
   public JavaScriptDimFilter(
+      // for backwards compatibility
       @JsonProperty("dimension") String dimension,
+      @JsonProperty("dimensions") String[] dimensions,
       @JsonProperty("function") String function
   )
   {
-    Preconditions.checkArgument(dimension != null, "dimension must not be null");
+    Preconditions.checkArgument(dimension != null ^ dimensions != null, "dimensions(xor dimension) must not be null");
     Preconditions.checkArgument(function != null, "function must not be null");
-    this.dimension = dimension;
+    this.dimensions = (dimensions != null) ? dimensions : new String[]{dimension};
     this.function = function;
   }
 
   @JsonProperty
-  public String getDimension()
+  public String[] getDimensions()
```

Comment:
i think this may cause serde errors given how Jackson works

need a serde check to make sure